tests/wasmbus.rs

```diff
@@ -281,40 +360,35 @@ async fn wasmbus() -> anyhow::Result<()> {
 
     let http_port = free_port().await?;
 
-    // NOTE: Links are advertised before the provider is started to prevent race condition, which
+    // NOTE(1): Links are advertised before the provider is started to prevent race condition, which
     // occurs if link is established after the providers starts, but before it subscribes to NATS
     // topics
-    let CtlOperationAck { accepted, error } = client
-        .advertise_link(
-            &actor_claims.subject,
-            &httpserver_provider_key.public_key(),
+    // NOTE(2): `kv-redis` provider fails to connect to Redis when only link-level configuration is
+    // given
+    try_join!(
+        assert_advertise_link(
+            &client,
+            &actor_claims,
+            &httpserver_provider_key,
             "wasmcloud:httpserver",
-            "default",
-            HashMap::from([(
-                "config_json".into(),
-                format!(r#"{{"address":"[{}]:{http_port}"}}"#, Ipv6Addr::UNSPECIFIED),
-            )]),
-        )
-        .await
-        .map_err(|e| anyhow!(e).context("failed to advertise link"))?;
-    ensure!(error == "");
-    ensure!(accepted);
-
-    let CtlOperationAck { accepted, error } = client
-        .advertise_link(
-            &actor_claims.subject,
-            &nats_provider_key.public_key(),
+            format!(r#"{{"address":"[{}]:{http_port}"}}"#, Ipv6Addr::UNSPECIFIED),
+        ),
+        assert_advertise_link(
+            &client,
+            &actor_claims,
+            &kvredis_provider_key,
+            "wasmcloud:keyvalue",
+            format!(r#"{{"url":"{redis_url}"}}"#),
+        ),
```

Comment:
I think, according to the provider code, `url` should be `URL` to properly parse as the KvRedisConfig